react/hooks/useAppSettings.ts

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from 'react-apollo'
+import { useEffect, useState } from 'react'
+
+import GET_SETTINGS from '../queries/getSettings.graphql'
+
+const DEFAULT_DECIMALS = 2
+const DEFAULT_PRICES_WITH_TAX = false
+
+interface Settings {
+  decimals: number
+  pricesWithTax: boolean
+}
+
+const useAppSettings = (): Settings => {
+  const [settings, setSettings] = useState<Settings>({
+    decimals: DEFAULT_DECIMALS,
+    pricesWithTax: DEFAULT_PRICES_WITH_TAX,
+  })
+
+  const { data, loading } = useQuery(GET_SETTINGS, { ssr: false })
+
+  useEffect(() => {
+    if (!loading) {
+      const { decimals, pricesWithTax } = JSON.parse(data?.appSettings?.message)
```

Comment:
Try to run `JSON.parse(undefined)` or `JSON.parse('')` on your browser. It will throw an error. You really have to check it before to make sure it won't throw a error due to foreseeable causes.

Reply:
Fair enough - Added better handling and an extra test for null values for piece of mind :) Cheers

Reply:
Again, I think you're getting the wrong picture here: if `data`, `appSettings`, or `message` are `null` or `undefined` or `''`, this will throw an error. You can't parse those values and even if you could, you couldn't deconstruct them.

Reply:
Sorry, long week... I see what you're saying, it should be OK now